### PR TITLE
tweak: Genesis provision tweaks

### PR DESCRIPTION
**Pull request by dhedey**
* Aligned genesis file formats
* Created a new enviroment variable name for the file, as it's different
* Fixed a bug with `genesis.use_olympia` resolution

### CODE REVIEW

File: cli-tools/src/main/java/com/radixdlt/cli/GenerateGenesis.java [hunk 1]
```diff
@@ -180,14 +183,45 @@ public static void main(String[] args) throws Exception {
...
-        writer.write(compressedGenesisDataBase64);
+      System.out.format(
+          "The base64-encoded genesis for use with RADIXDLT_GENESIS_DATA / network.genesis_data"
+              + " is:%n%s%n",
+          compressedGenesisDataBase64);
+      System.out.println();
+      var filePath = new File("genesis_data.bin").getAbsolutePath();
+      System.out.format(
+          "Also saving the genesis data file in binary format for use with"
+              + " RADIXDLT_GENESIS_DATA_FILE / network.genesis_data_file to: %s%n",
```
Comment (jakrawcz-rdx) [inline cli-tools/src/main/java/com/radixdlt/cli/GenerateGenesis.java:198]: (minor) In CLI-related code, I find the multiline strings handy (`"""`)

File: cli-tools/src/main/java/com/radixdlt/cli/GenerateGenesis.java [hunk 2]
```diff
@@ -180,14 +183,45 @@ public static void main(String[] args) throws Exception {
...
+    if (asName.isPresent()) {
+      return asName.get();
+    }
+    try {
+      final var asId = Network.ofId(Integer.parseInt(commandLineValue));
+      if (asId.isPresent()) {
+        return asId.get();
+      }
+      throw new RuntimeException("Was not a valid network id");
+    } catch (Exception ex) {
```
Comment (jakrawcz-rdx) [inline cli-tools/src/main/java/com/radixdlt/cli/GenerateGenesis.java:217]: hmm, something must have went wrong with the exceptions in this method (I mean: the `catch` 100% catches the `RuntimeException` thrown above)

File: cli-tools/src/main/java/com/radixdlt/cli/GenerateGenesis.java [hunk 3]
```diff
@@ -180,14 +183,45 @@ public static void main(String[] args) throws Exception {
...
     }
   }
 
+  private static Network parseNetwork(String commandLineValue) {
+    final var asName = Network.ofName(commandLineValue);
+    if (asName.isPresent()) {
+      return asName.get();
+    }
+    try {
+      final var asId = Network.ofId(Integer.parseInt(commandLineValue));
```
Comment (jakrawcz-rdx) [inline cli-tools/src/main/java/com/radixdlt/cli/GenerateGenesis.java:212]: if the `Integer.parseInt` is the reason for the weird `throw/catch`, then may I recommend `Ints.tryParse()`?

File: core/src/main/java/com/radixdlt/bootstrap/GenesisFileStore.java [hunk 4]
```diff
@@ -69,76 +69,104 @@
...
-        if (!file.createNewFile()) {
-          throw new RuntimeException("Genesis file doesn't exist and failed to create");
+        if (!genesisFolder.mkdirs()) {
+          throw new RuntimeException("Genesis data folder doesn't exist and failed to create");
+        }
+        if (!genesisDataFile.createNewFile()) {
+          throw new RuntimeException("Genesis data file doesn't exist and failed to create");
+        }
+        if (!genesisHashFile.createNewFile()) {
+          throw new RuntimeException("Genesis hash file doesn't exist and failed to create");
```
Comment (jakrawcz-rdx) [inline core/src/main/java/com/radixdlt/bootstrap/GenesisFileStore.java:103]: do they need to be explicitly created like that? won't `FileOutputStream` (re-)create them anyways?

**Review by kofrdx — APPROVED** [review]

**Review by jakrawcz-rdx — APPROVED** [review]
I understand the one-off nature of this code, so all my comments are optional 👍

File: cli-tools/src/main/java/com/radixdlt/cli/GenerateGenesis.java (hunk 2 above)
Comment (dhedey) [reply cli-tools/src/main/java/com/radixdlt/cli/GenerateGenesis.java:217]: This was actually intentional - I wanted a wrapped error! Or more specifically, I wanted all the different errors that could be thrown to all be wrapped in an error with the same helpful error message.

Comment (dhedey) [reply cli-tools/src/main/java/com/radixdlt/cli/GenerateGenesis.java:217]: But based on your suggestion to use `Ints.tryParse()` I have gone full functional and flattened all this into 4 lines and avoided the gnarly errors 👍  :)

**Pull request merged**